src/main/java/com/legendary/coffeeShop/dao/repositories/ComponentRepository.java

```diff
@@ -0,0 +1,11 @@
+package com.legendary.coffeeShop.dao.repositories;
+
+import com.legendary.coffeeShop.dao.entities.Component;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+import java.util.Set;
+
+public interface ComponentRepository extends JpaRepository<Component, Long> {
+    Set<Component> findByProductTypesProductTypeContains(String type);
+    Component findByNameContains(String componentName);
```

Comment:
Please change the component identify from "name" to the new column "type" (the name will represent the display name at the client side and the "type" will represent the component identify so the search at the DB should be by the unique value and without any whitespaces...)

Reply:
Please also add the search by status